### PR TITLE
added blog/content/post/2019-04-12-bit-bang-theory-with-rte.md an…

### PR DESCRIPTION
**Pull request by lwcislo**
…d add blog/static/covers/bit_bang_cover.jpg

### CODE REVIEW

File: blog/content/post/2019-04-12-bit-bang-theory-with-rte.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+This technique is called **The Bit Banging**.
+
+So, let's assume, that we have some board with a chip, i.e. STM32 series, which
+is very popular, and a binary image, which we want to be flashed on it.
+To do it, at the very beginning we need some software that provides us a way to
+manipulate the state of RTE pins as if they were pins of a programmer. As we
+prefer open source we used OpenOCD (Open-On-Chip_Debbuger). This is a
+well-developed tool for such jobs, but, unfortunately, it doesn't support
+Orange Pi Zero. And this is our microcomputer assembled with RTE.
```
Comment (artur-rs) [inline blog/content/post/2019-04-12-bit-bang-theory-with-rte.md]: attached is better word

File: blog/content/post/2019-04-12-bit-bang-theory-with-rte.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+targets
+reset halt
+program vitroio-node-1.1.1-demo-dht.bin verify 0x8000000
+reset
+shutdown
+```
+
+Which means :
+
+Take interface config file for Orange Pi,
```
Comment (artur-rs) [inline blog/content/post/2019-04-12-bit-bang-theory-with-rte.md]: @lwcislo I think that enumerating all steps would be better

File: blog/content/post/2019-04-12-bit-bang-theory-with-rte.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+set speed,
+start,
+find suitable connected chip,
+stop that chip,
+flash with file (which is in our directory `~/bootloader/`) starting at
+0x8000000 address then verify if flashig was successful,
+reset device,
+close bit banging procedure.
+
+Then we typed openocd in bootloader directory. There is no need to add any more,
```
Comment (artur-rs) [inline blog/content/post/2019-04-12-bit-bang-theory-with-rte.md]: \`openocd\`

File: blog/content/post/2019-04-12-bit-bang-theory-with-rte.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+in procedure 'ocd_process_reset'
+in procedure 'ocd_process_reset_inner' called at file "embedded:startup.tcl",
+line  244
+ ```
+
+But in openocd documentation, this is described as more or less irrelevant.
+All in all our microchip has been flashed, and this action has been verified.
+
+## Summary
+
```
Comment (artur-rs) [inline blog/content/post/2019-04-12-bit-bang-theory-with-rte.md]: I think that here You should sum up the whole article in few sentences :)

**Comment by lwcislo** [comment]
@artur-rs @pietrushnic  Moved PR to https://github.com/3mdeb/news-and-ideas/pull/73/

**Pull request closed without merging**